### DISCUSSION TITLE
Renovate bot consistently  has Blocked PR with no reason

### DISCUSSION THREAD

**Discussion by Harris-Miller** [renovatebot/renovate#12011] [Request Help]
We have Gitlab installed on our intranet and have a Renovate-Bot repo that is set to run nightly on a CI/CD schedule. It touches 24 other repos. All these repos have `renovate.json` files that we have run through the validator and know are correct. They also all extend a `default.json` that lives in the same Renovate-Bot previously mentioned.

Most of the reps renovate bot run without issue. However, after recently upgrading from `^21` to `^26` (our infrastructure isn't set up for `node@^14` yet, so we haven't bump to `^27` yet), we found that a few repos consistently just print out
```
INFO: Blocked PR (repository=**redacted**)
       "prBlockedBy": undefined,
       "prNo": undefined,
       "result": "no-work"
```
We don't know what is going on and haven't found anything in your docs or on the internet in general about this.

How can I better see what is going on? Can we just reset renovate-bot for the repos in question? (eg, remove all the open MRs and delete all the created branches)

We're at a lost on how to fix this

**Comment by rarkins** [comment]
Run with debug logs to see more info

**Comment by Harris-Miller** [comment]
@rarkins Ran again with `LOG_LEVEL=debug`. However, that did not give me any insight into why there are blocked PRs.

So let me give you a scenario. Let's say package A depends on package B, Both are internal packages. We did a new release of package B, and had an open MR for package A which includes the bump of package B and additional work. The MR for package A sat for a few days because of revisions. Meanwhile, renovate bot created an MR on package A for package B, and it was not auto-merged because the gitlab pipeline failed. We know not to close it ourselves (as we don't want to block future MRs for package B), so we just let it sit there, knowing renovate bot would close it itself. We finally merged our MR in package A which includes the bump to package B

However, the renovate bot for package B is still there. Furthermore, we have clicked "If you want to rebase/retry this MR, check this box" and that has remained clicked for days now. renovate bot is not rebasing it or closing it.

Like I first mentioned, the big change was we bumped from `renovate@^21` to `renovate@^26`. We also made sure that all our configs we migrated.

The DEBUG output that you asked me to do shows nothing for package B, aside from it being listed out as a dependency as part of DEBUG: `packageFiles with updates (repository=**redacted**)`

We have 10 of these in the print out still
```
INFO: Blocked PR (repository=**redacted**)
       "prBlockedBy": undefined,
       "prNo": undefined,
       "result": "no-work"
```
We don't understand why renovate bot is not rebasing or closing that MR. I should also mention that the dependency dashboard does not show the MR in the "Open" list, but it does show it in the "Related merge requests" section below it

**Reply by rarkins** [reply]
> So let me give you a scenario. Let's say package A depends on package B, Both are internal packages. We did a new release of package B, and had an open MR for package A which includes the bump of package B and additional work. 

Do you mean it did or did not include the "new release of package B"? Try not to mix terms like "new release" and "the bump".

> The MR for package A sat for a few days because of revisions. Meanwhile, renovate bot created an MR on package A for package B, and it was not auto-merged because the gitlab pipeline failed. We know not to close it ourselves (as we don't want to block future MRs for package B), 

Future MRs should only be blocked if it's the exact same version of package B. e.g. you can close a failing MR for 1.0.2 and then when 1.0.3 is released you get a new PR. Now if the new version was `2.0.0` (a major update) and you closed it then yes, you'd ignore all v2 updates including a future `2.0.1`. You can rename an ignored MR at any time to remove that blocking though

> so we just let it sit there, knowing renovate bot would close it itself. 

Why would it close it?

> We finally merged our MR in package A which includes the bump to package B

Again please clarify "the bump". Maybe use versions in your examples.

> However, the renovate bot for package B is still there. 

I do not understand this sentence

> Furthermore, we have clicked "If you want to rebase/retry this MR, check this box" and that has remained clicked for days now. renovate bot is not rebasing it or closing it.

You should check your logs specifically for this.

**Reply by Harris-Miller** [reply]
> Do you mean it did or did not include the "new release of package B"? Try not to mix terms like "new release" and "the bump".

Package A included the semver bump for the new release of Package B. eg B went from `2.1.0` to `2.2.0` and the MR for package A included this semver bump

> Future MRs should only be blocked if it's the exact same version of package B. e.g. you can close a failing MR for 1.0.2 and then when 1.0.3 is released you get a new PR. Now if the new version was 2.0.0 (a major update) and you closed it then yes, you'd ignore all v2 updates including a future 2.0.1. You can rename an ignored MR at any time to remove that blocking though

I will try that

> Why would it close it?

We have seen this happening the past, at least that's why I remember it happen. Renovate would close stale MRs, does it not?

> Again please clarify "the bump". Maybe use versions in your examples.

I mean "bump in semver", eg `2.1.0` -> `2.2.0`

> I do not understand this sentence

That should have said "However, the renovate bot **MR** for package B is still there.

> You should check your logs specifically for this.

I will check and get back to you

**Reply by rarkins** [reply]
Renovate doesn't close stale MRs

**Comment by Harris-Miller** [comment]
> You can rename an ignored MR at any time to remove that blocking though

@rarkins I will try that as that will likely fix what I'm trying to do. Will get back to you in a day or two. Thank you for your help so far

**Comment by Harris-Miller** [comment]
@rarkins I narrowed down the issue to rate limiting, add `:disableRateLimiting` to our configs allowed all the PRs to be created as expected.

We are no longer seeing
```
INFO: Blocked PR (repository=**redacted**)
       "prBlockedBy": undefined,
       "prNo": undefined,
       "result": "no-work"
```
in the renovate output. I'm going to assume that `"prBlockedBy": undefined` is a bug on your end and it should be something to the effect of `"prBlockedBy": "rate limiting"`

**Reply by Harris-Miller** [reply]
Also, renaming ignored MRs did help and those packages are now being updated as expected
